src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> t = tail;
+        Node<T> newNode = new Node<>(t, value, null);
+        tail = newNode;
+        if (t != null) {
+            t.next = newNode;
+        } else {
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            head.prev = new Node<>(null, value, head);
+            head = head.prev;
+            size++;
+        } else {
+            Node<T> node = getNodeByIndex(index);
+            Node<T> current = new Node<>(node.prev, value, node);
+            current.next.prev = current;
+            current.prev.next = current;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNodeByIndex(index);
+        T element = node.item;
+        node.item = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> currentNode = getNodeByIndex(index);
+        T removedValue = currentNode.item;
+        unlink(currentNode);
+        if (size > 0) {
+            size--;
+        }
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            if (current.item == null && object == null
+                    || current.item != null && current.item.equals(object)) {
+                remove(i);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlink(Node<T> node) {
+        if (node.next == null) {
+            tail = node.prev;
+            if (node.prev != null) {
+                node.prev.next = null;
+            }
+        } else {
+            node.next.prev = node.prev;
+        }
+        if (node.prev == null) {
+            head = node.next;
+            if (node.next != null) {
+                node.next.prev = null;
+            }
+        } else {
+            node.prev.next = node.next;
+        }
```

Comment:
Quite complicated, you can unlink by using only 2 `if-else` blocks without inner `if`s. Also, `size` adjustment is belongs to `unlink` method, since when you `unlink` your `size` changes immediately.